### PR TITLE
[api][SIMS #279] Create education porgrams

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Partial delivery of SIMS #279.

- Created migrations to create the table education_programs;
- Created the education programs controller with the method to create a new education program;

### CODE REVIEW

File: sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql [hunk 1]
```diff
@@ -0,0 +1,64 @@
+CREATE TABLE IF NOT EXISTS education_programs(
+  id SERIAL PRIMARY KEY,
+  program_name VARCHAR(300) NOT NULL,
+  program_description VARCHAR(300) NOT NULL,
```
Comment (ann-aot) [inline sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql:4]: will 300 be enough for the description?

File: sources/packages/api/src/database/entities/index.ts [hunk 2]
```diff
@@ -8,3 +8,4 @@ export * from "./institution-user.model";
 export * from "./institution-user-auth.model";
 export * from "./sequence-controls.model";
 export * from "./institution-location.model";
+export * from "./education-program.model";
```
Comment (guru-aot) [inline sources/packages/api/src/database/entities/index.ts:11]: Please rename to education-programs.model

File: sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  sabc_code VARCHAR(50) NULL,
+  regulatory_body VARCHAR(100) NOT NULL,
+  delivered_on_site BIT NOT NULL, 
+  delivered_online BIT NOT NULL,
+  delivered_online_also_onsite VARCHAR(50) NULL, 
+  same_online_credits_earned VARCHAR(50) NULL,
+  earn_academic_credits_other_institution VARCHAR(50) NULL,
+  course_load_calculation VARCHAR(50) NOT NULL,
+  average_hours_study INT NOT NULL,
+  completion_years VARCHAR(50) NOT NULL,
```
Comment (ann-aot) [inline sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql:18]: `year`, will it be better to use INTEGER

Comment (guru-aot) [reply sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql:18]: I feel it is okay as per the UI the values are not integers, it like 

- 12weeksOrLessThen1year
- 2yearsToLessThan3years
- 3yearsToLessThan4years
etc..

**Review by guru-aot — APPROVED** [review]
LGTM

File: sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql:4]: I checked with @abschwenker and @alpivett and we are going to increase to 500.

File: sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql:18]: As per @guru-aot mention, we are going to store something different than a number.

**Review by ann-aot — APPROVED** [review]
LGTM

File: sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src-sql/EducationPrograms/Create-education-programs.sql]: Updated to 500.

**Review by abschwenker — APPROVED** [review]

**Pull request merged**